tools/summaries/build_template_summaries.py

```diff
@@ -165,26 +169,77 @@ def build_template_summaries(place_dcids: List[str], stat_var_json=str) -> Dict:
   return summaries
 
 
-def main():
-  with open("priority-places.json") as f:
-    priority_places = json.load(f)
-
-  with open("places-countries.json") as f:
-    countries = json.load(f)
-
-  priority_summaries = build_template_summaries(place_dcids=priority_places,
-                                                stat_var_json=_STAT_VAR_JSON)
-
-  country_summaries = build_template_summaries(place_dcids=countries,
-                                               stat_var_json=_STAT_VAR_JSON)
-
-  all_summaries = utils.combine_summaries(
-      [priority_summaries, country_summaries])
-
-  utils.write_summaries_to_file(summaries=all_summaries,
-                                output_file=_OUTPUT_FILENAME)
+def build_template_summaries_for_sitemap(
+    sitemap_url: str,
+    stat_var_json: str = _STAT_VAR_JSON,
+    batch_size: int = _BATCH_SIZE,
+    output_file: str = _OUTPUT_FILE) -> Dict:
+  """Generate summaries for all places in a sitemap"""
+  start_time = time.time()
+
+  # Extract places to create summaries for from sitemap
+  places = utils.get_places_from_sitemap(sitemap_url)
+
+  # Get summaries in batched calls
+  batch_num = 0
+  batch_start_time = time.time()
+  for batch in utils.batched(places, batch_size):
+    logging.info(f'Processing batch number {batch_num}')
+    summaries = build_template_summaries(place_dcids=batch,
+                                         stat_var_json=stat_var_json)
+    # Write intermediate results to a temporary file
+    # This allows us to save partial progress incase we hit server errors
+    temp_path = _TEMP_FILENAME.format(num=batch_num)
+    utils.write_summaries_to_file(summaries=summaries, output_file=temp_path)
+    logging.info(f'Wrote intermediate results to {temp_path}')
+    logging.info(
+        f'Took {timedelta(seconds=time.time()-batch_start_time)} to write batch'
+    )
+    batch_num += 1
+    batch_start_time = time.time()
+
+  # Combine intermediate results into one
+  all_data = []
+  for i in range(batch_num):
+    filename = _TEMP_FILENAME.format(num=i)
+    with open(filename) as f:
+      all_data.append(json.load(f))
+
+  # Write summary
+  summaries = utils.combine_summaries(all_data)
+  utils.write_summaries_to_file(summaries, output_file)
+
+  # Cleanup temp files
+  for i in range(batch_num):
+    filename = _TEMP_FILENAME.format(num=i)
+    os.remove(filename)
+
+  logging.info('Done!')
+  logging.info(
+      f'Total elapsed time: {timedelta(seconds=time.time()-start_time)}')
+
+
+@click.command()
+@click.argument('sitemap_url')
```

Comment:
Should this be a path to a local file instead? the County.0.txt and Country.0.txt are already in the repo, and pointing at a local file could be helpful for testing the script

Reply:
This is a good idea. Switched this function and underlying helper function to reading a filepath instead of a URL. Also updated the ./run script to use filepaths.